pkgs/applications/misc/st/default.nix

```diff
@@ -4,30 +4,28 @@
 with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  name = "st-0.6";
+  name = "st-0.7";
   
   src = fetchurl {
     url = "http://dl.suckless.org/st/${name}.tar.gz";
-    sha256 = "0avsfc1qp8zvshsfjwwrkvk411jlqy58z225bsdhjkl1qc40qcc5";
+    sha256 = "f7870d906ccc988926eef2cc98950a99cc78725b685e934c422c03c1234e6000";
   };
-
-  inherit patches;
+  
+  patches = with self; [ ./st-fix-deletekey.patch ];
```

Comment:
This still does not apply the old patches. The `with self;` only brings the attributes of`self` into scope, in fact there *is* no `self` in scope but this doesn't fail because the list doesn't reference any variables that might belong to self. You probably want to do something like `let patches' = patches; in` outside of the set and then have `patches = patches' ++ [ ./st-fix-deletekey.patch ]` here